### PR TITLE
feat(snackbar): Implement feedback from Design

### LINKED ISSUE #4156: Snackbar: Designer feedback
Design has requested the following changes to the new Snackbar (PR #4091):

- [ ] 100% width on "mobile-sized" screens
- [ ] Remove `mdc-snackbar--wide` class and automatically apply wide margins w/ `@media` queries
- [ ] Remove Sass variables for animation delays
    (The motion spec has a 150ms delay for exit animations, but according to Design, that's a mistake; they should both always be `0ms`)
    * `$mdc-snackbar-enter-delay`
    * `$mdc-snackbar-exit-delay`

### PR DESCRIPTION
**Pull request by acdvorak**
Fixes #4156

### CODE REVIEW

File: packages/mdc-snackbar/_mixins.scss [hunk 1]
```diff
@@ -45,9 +45,15 @@
   }
 }
 
-@mixin mdc-snackbar-min-width($min-width) {
+@mixin mdc-snackbar-min-width($min-width, $mobile-breakpoint: $mdc-snackbar-mobile-breakpoint) {
   .mdc-snackbar__surface {
     min-width: $min-width;
+
+    // The first media query ensures that snackbars are always 100% width on mobile devices, as required by the spec.
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply packages/mdc-snackbar/_mixins.scss]: They might have different units.

E.g., somebody could pass `30em` to `min-width` and leave the media query in `px`.

File: packages/mdc-snackbar/README.md [hunk 2]
```diff
@@ -149,8 +140,8 @@ Mixin | Description
 `mdc-snackbar-fill-color($color)` | Sets the fill color of the snackbar.
 `mdc-snackbar-label-ink-color($color)` | Sets the color of the snackbar's label text.
 `mdc-snackbar-shape-radius($radius, $rtl-reflexive)` | Sets the rounded shape to snackbar surface with given radius size. Set `$rtl-reflexive` to true to flip radius values in RTL context, defaults to false.
-`mdc-snackbar-min-width($min-width)` | Sets the minimum width of the snackbar (defaults to `344px`).
-`mdc-snackbar-max-width($max-width)` | Sets the maximum width of the snackbar (defaults to `672px`).
+`mdc-snackbar-min-width($min-width)` | Sets the `min-width` of the surface on tablet/desktop devices. On mobile, the width is automatically set to 100%. To specify a custom breakpoint, override the `$mdc-snackbar-mobile-breakpoint` variable before calling this mixin.
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply packages/mdc-snackbar/README.md]: Done

File: test/screenshot/golden.json [hunk 3]
```diff
@@ -1150,102 +1150,93 @@
     }
   },
   "spec/mdc-snackbar/classes/baseline-with-action.html": {
-    "public_url": "https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/12/05/21_03_57_990/spec/mdc-snackbar/classes/baseline-with-action.html?utm_source=golden_json",
+    "public_url": "https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/12/10/22_24_22_755/spec/mdc-snackbar/classes/baseline-with-action.html?utm_source=golden_json",
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply test/screenshot/golden.json]: Reverted. Good catch!

File: packages/mdc-snackbar/README.md [hunk 4]
```diff
@@ -82,6 +82,16 @@ Action buttons with long text should be positioned _below_ the label instead of
 </div>
 ```
 
+Alternatively, you can call the `mdc-snackbar-layout-stacked` mixin from Sass:
+
+```scss
+@media (min-width: $mdc-snackbar-mobile-breakpoint) {
```
Comment (kfranqueiro) [inline packages/mdc-snackbar/README.md]: max-width would probably be more appropriate for this example?

Comment (acdvorak) [reply packages/mdc-snackbar/README.md]: Done

**Pull request merged**